src/HotChocolate/ApolloFederation/test/ApolloFederation.Tests/Directives/ExternalDirectiveTests.cs

```diff
@@ -29,6 +31,57 @@ public void AddExternalDirective_EnsureAvailableInSchema()
             Assert.Equal(0, directive.Arguments.Count());
             Assert.Collection(directive.Locations,
                 t => Assert.Equal(DirectiveLocation.FieldDefinition, t));
-        }        
+        }
+
+        [Fact]
+        public void AnnotateExternalToObjectFieldCodeFirst()
+        {
+            // arrange
+            // act
+            var schema = Schema.Create(
+                t =>
+                {
+                    t.RegisterQueryType(new ObjectType(
+                        o => o.Name("Query")
+                            .Field("field")
+                            .Argument("a", a => a.Type<StringType>())
+                            .Type<StringType>()
+                            .External()
+                    ));
+
+                    t.RegisterDirective<ExternalDirectiveType>();
+
+                    t.Use(next => context => default(ValueTask));
+                });
+
+            ObjectType query = schema.GetType<ObjectType>("Query");
```

Comment:
where is the assert part of this test?

Reply:
The test should always have proper asserts ... Assert.Collection for instance

Reply:
As to why, a single exception is not explicit enough and might be more difficult to read in a test result file.

Reply:
Replaced the collectin.single with Assert.Collection